modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -1004,21 +1069,24 @@ final BufferedImage test_getPixelsIm() {
 
     private class HostContainer implements HostInterface {
 
+        // These methods are invoked from FX thread
+
         @Override
         public void setEmbeddedStage(EmbeddedStageInterface embeddedStage) {
             stagePeer = embeddedStage;
-            if (stagePeer == null) {
+            var hStagePeer = stagePeer;
+            if (hStagePeer == null) {
```

Comment:
If this is always invoked from the FX thread, you don't need to locally capture the peers.